qiskit/algorithms/gradients/lin_comb_estimator_gradient.py

```diff
@@ -124,7 +151,7 @@ def _run(
 
         gradients = []
         for i, result in enumerate(results):
-            gradient_ = np.zeros(len(metadata_[i]["parameters"]))
+            gradient_ = np.zeros(len(metadata_[i]["parameters"]), dtype="complex_")
```

Comment:
Does this need to be complex. Aren't the results from estimator always going to be real values for the expectations?

Reply:
Yes, it needs to be complex. When we pass observables `X-iY` (i is an imaginary coefficient), which is used in LinCombQFI the return value will be complex.